app/models/customer.rb

```diff
@@ -2,9 +2,14 @@
 
 class Customer < ApplicationRecord
   has_many :invoices, dependent: :destroy
+  has_many :transactions, through: :invoices
+  has_many :invoice_items, through: :invoices
+  has_many :items, through: :invoice_items
```

Comment:
Don't really agree that there's a has_many relationship here... how do customers have many items?